docker-compose.yml

```diff
@@ -16,24 +16,68 @@ services:
     image: postgis/postgis:13-3.2-alpine
     volumes:
       - pg_data:/var/lib/postgresql/data
+      # copy the sql script to create db/tables
+      - ./sql/init_db.sql:/docker-entrypoint-initdb.d/init_db.sql
     environment:
-      - POSTGRES_PASSWORD=postgres
-      - POSTGRES_USER=postgres
+      POSTGRES_PASSWORD: postgres
+      POSTGRES_USER: postgres
+    restart: always
     ports:
       - 5432:5432
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U postgres"]
+      start_period: 60s
+      interval: 5s
+      timeout: 3s
+      retries: 10
 
   api_service:
     container_name: api_service
     build:
         context: .
-        dockerfile: Dockerfile
+        dockerfile: Dockerfile.ecs
     image: wcs/mermaid-api:latest
-    volumes:
-      - $PWD/src:/var/projects/webapp
     ports:
-      - 8080:8080
+      - 8081:8081
+    environment:
+      # db
+      DB_NAME: mermaid
+      DB_HOST: api_db
+      DB_PORT: 5432
+      DB_USER: postgres
+      DB_PASSWORD: postgres
+
+      # Auth0
+      MERMAID_DOMAIN: datamermaid.auth0.com
+      MERMAID_API_AUDIENCE: https://dev-api.datamermaid.org
```

Comment:
This is because we're deploying immutable architecture. We don't want to be editing servers "on the fly". Here is a good article: https://www.digitalocean.com/community/tutorials/what-is-immutable-infrastructure

This will help understand why this approach is better and why companies are adopting it more and more...

Reply:
Thanks for the link. I think I get the immutable thing, and in fact that's the mode we've been in for some time. I think what I'm not understanding is how `docker-compose.yml` is used. Here, it looks to me like when CI/CD runs, it will hard-code env vars in the container to dev versions; what happens on prod? In our current setup, of course, we're only using `docker-compose.yml` locally; but my understanding is that this will be part of the GH action? But I may be wrong about that.

Reply:
Yes, env vars get set in the container (just like they do now in EB). This will be the same for prod. docker-compose is only for local (and perhaps CI tests - M1700).